apps/dotcms-ui-e2e/src/pages/DotTemplates/DotTemplates.ts

```diff
@@ -0,0 +1,75 @@
+import Form from '../../support/shared/Form';
+import Navigation from '../../support/shared/Navigation';
+import Page from '../../support/shared/Page';
+
+const URL = '/dotAdmin/#/templates';
+const CREATE_TEMPLATE_BUTTON = 'dot-action-button dot-icon i';
+
+class DotTemplates {
+    static openPage() {
+        cy.intercept('GET', 'v1/templates').as('templates');
+        Navigation.visit(URL);
+        Navigation.assertPageUrlIs(URL);
+        cy.wait('@templates');
+    }
+
+    static checkTemplatesPageLoaded() {
+        Page.assertElementContains('.p-breadcrumb > ul > :nth-child(3)', `Templates`); // Header
+        Page.assertElementSize('p-table', 1); // Table
+        Page.assertElementSize('.action-header__global-search input[type="text"]', 1); // Global Search Input
+        Page.assertElementContains('.template-listing__header-options p-checkbox', `Show Archived`); // Show Archived button
+        Page.assertElementContains('.template-listing__header-options .p-button-label', `Actions`); // BulkAction button
+        Page.assertElementContains(CREATE_TEMPLATE_BUTTON, `add`); // Action button
+    }
+
+    static openCreateTemplateDialog() {
+        Page.click(CREATE_TEMPLATE_BUTTON);
+        Page.assertElementContains('.p-dialog-title', 'Create a template');
+        Page.assertElementContains('label[data-testid="designer"] span', 'Designer');
+        Page.assertElementContains('label[data-testid="advanced"] span', 'Advanced');
+        Page.assertElementContains(
+            'p[data-testid="description"]',
+            'Template Designer create reusable templates using a drag and drop interface and tools.'
+        );
+        Page.assertElementContains('dot-dot-template-selector button', 'Next').click();
+    }
+
+    static fillCreateTemplateForm({ title, theme }: { title: string; theme: string }) {
+        // TODO: Complete to fill extra fields (not required)
+        Form.fill('input[data-testid="templatePropsTitleField"]', title);
+        Page.click('dot-theme-selector-dropdown[data-testid="templatePropsThemeField"]');
+        Page.click(
+            cy.get('.theme-selector__data-list-item .dot-theme-item__label').contains(theme)
+        );
+    }
+
+    static submitCreateTemplateForm() {
+        Page.click('button[data-testid="dotFormDialogSave"');
+    }
+
+    static checkEditTemplatesPageLoaded() {
+        // TODO: Check other components loaded on Edit Template page
+        Page.assertElementContains('#p-tabpanel-0-label .p-tabview-title', `design`); // Secondary Menu bar
+        Page.assertElementContains('#p-tabpanel-1-label .p-tabview-title', `Permissions`); // Secondary Menu bar
+        Page.assertElementContains('#p-tabpanel-2-label .p-tabview-title', `History`); // Secondary Menu bar
+        Page.assertElementContains('.dot-edit-layout__toolbar-save', `Save`); // Save button
+        Page.assertElementContains('dot-searchable-dropdown button', `Add a Container`); // Add container select
```

Comment:
Not sure why we want to check all these elements to assert that the page is loaded. What's the reasoning behind this practice?

Reply:
not necessary, but is a good practice make sure every one of the elements is loaded before continue

Reply:
https://github.com/dotCMS/core-web/pull/1621/files#r593493939